server/modules/activityLog/manager/activityLogManager.js

```diff
@@ -81,9 +82,40 @@ const _getAvailableActivityTypes = async (surveyId, user) => {
   )(user)
 }
 
+const _canUserAccessSurvey = async (user, surveyUuid) => {
+  const authGroups = await AuthGroupRepository.fetchUserGroups(User.getUuid(user))
+  const userSurveyAuthGroups = R.filter(g => AuthGroup.getSurveyUuid(g) === surveyUuid)(authGroups)
+  return !R.isEmpty(userSurveyAuthGroups)
+}
+
+const _transformActivityLogUser = surveyUuid => async activityLogDb => {
+  const user = await UserRepository.fetchUserByUuid(ActivityLog.getContentUuid(activityLogDb))
+  const canUserAccessSurvey = await _canUserAccessSurvey(user, surveyUuid)
+
+  // assoc user email and name only if user has not been removed from survey
+  return R.when(
+    R.always(canUserAccessSurvey),
+    R.pipe(
+      R.assocPath([ActivityLog.keys.content, ActivityLog.keysContent.userEmail], User.getEmail(user)),
+      R.assocPath([ActivityLog.keys.content, ActivityLog.keysContent.userName], User.getName(user)),
+    ),
+    activityLogDb
+  )
+}
+
 export const fetch = async (user, surveyId, offset, limit) => {
-  const activityTypes = await _getAvailableActivityTypes(surveyId, user)
-  return await ActivityLogRepository.fetch(surveyId, activityTypes, offset, limit)
+  const surveyInfo = await SurveyRepository.fetchSurveyById(surveyId)
+  const surveyUuid = Survey.getUuid(surveyInfo)
+  const activityTypes = await _getAvailableActivityTypes(surveyUuid, user)
+  const activityLogsDb = await ActivityLogRepository.fetch(surveyId, activityTypes, offset, limit)
+
+  return await Promise.all(activityLogsDb.map(async activityLogDb => {
+    if (R.includes(ActivityLog.getType(activityLogDb), [ActivityLog.type.userInvite, ActivityLog.type.userUpdate])) {
+      return await _transformActivityLogUser(surveyUuid)(activityLogDb)
+    } else {
+      return activityLogDb
+    }
+  }))
 }
```

Comment:
can't you get this with a join with user, auth_group and auth_group_user tables?

Reply:
yes, right